api/database.py

```diff
@@ -56,19 +73,38 @@ def init():
     # data = data.sample
     # data = data[lambda row: (row.Tumor_type_code == 'BRCA') | (row.Tumor_type_code == 'COAD')]
 
+    # Remove those rows that do not have this data
     data['T'] = data['Time_Diagnosis_Last_followup']
+    data['T'].astype(str)
+
+    # Rename columns to used names
+    if dataset != './SIM.csv':
+        data['MSI_ARTUR'] = data['MSI']
+        data['clinical_stage'] = data['clinicl_stge']
+        data['Anatomical_location'] = data['Atomical_location']
+        data['T'] = data['Time_Diagnosis_Last_followup']
+
+    # Assumption> fill missing if undefined
+    remove_nas = lambda x: "missing" if pd.isna(x) else x
+    for attr in ["PreOp_treatment_yesno", "PostOp_type_treatment"]:
+        data[attr] = data[attr].map(remove_nas)
+
+    # Fill consistent spelling for yes/no cells
+    to_lower = lambda x: x.lower() if isinstance(x, str) and x.lower() in ["yes", "no"] else x
+    for attr in ["PreOp_treatment_yesno", "PostOp_type_treatment", "Neuralinv", "Vascinv"]:
+        data[attr] = data[attr].map(to_lower)
+
+    data['Anatomical_location'].astype(str)
+
     data['E'] = data['Event_last_followup'] == 'Dead'
+    data['E'].astype(bool)
 
     tumor_types = uniq(data.Tumor_type_code)
     cell_types = uniq(c for c in data.columns if 'TUMOR' in c or 'STROMA' in c)
 
     dfs = []
     for t in tumor_types:
-        df = data_per_type(
-          data[
-            (data.Tumor_type_code == t) &
-            (data['PreOp_treatment_yesno'] == 'No')],
-          cell_types)
+        df = data_per_type(data[(data.Tumor_type_code == t)], cell_types)
```

Comment:
I guess you talked with them, but I think they wanted to have only patients where the `PreOp_treatment_yesno` is set no. But maybe they changed their mind?

Reply:
That column had lots of undefined values so I think I removed it because otherwise the number of datapoints would be very small. But we can try and see!